pkgs/cronet_http/example/lib/main.dart

```diff
@@ -73,9 +75,13 @@ class _HomePageState extends State<HomePage> {
     }
 
     final books = await _findMatchingBooks(query);
-    setState(() {
-      _books = books;
-    });
+    if (query == _lastQuery) {
+      // Avoid the situation where a slow-running query finishes late and
+      // replaces newer search results.
+      setState(() {
+        _books = books;
+      });
+    }
```

Comment:
[optional] consider flipping the condition and using an early return guard clause

```suggestion
    // Avoid the situation where a slow-running query finishes late and
    // replaces newer search results.
    if (query != _lastQuery) return;
    setState(() {
      _books = books;
    });
```